src/openrct2-ui/windows/TopToolbar.cpp

```diff
@@ -1377,8 +1377,11 @@ static void sub_6E1F34(
                 {
                     int16_t z = gSceneryCtrlPressZ;
 
-                    screen_get_map_xy_quadrant_with_z(x, y, z, grid_x, grid_y, &cl);
-
+                    auto mapCoords = screen_get_map_xy_quadrant_with_z({ x, y }, z, &cl);
+                    if (!mapCoords)
```

Comment:
You will need to set *grid_x to `LOCATION_NULL` to propagate the problem upwards.